### DISCUSSION TITLE
I am looking for the build artifacts location in GoCD server, also my pipelines are very slow.

### DISCUSSION THREAD

**Discussion by ghost** [gocd/gocd#11594] [Q&A]
I am looking for the build artifacts location in GoCD server, also my pipelines are very slow.

npm build took long time to finished. When I run 3-5 pipeline build server hanged using 64GB RAM + 400 GB memory (Windows Server 2019)

I used ci cmd instead of install change the firewall settings

**Comment by chadlwilson** [comment]
Sorry, it's not really clear what you're asking here, or what you have already done to investigate.

Artifact files storage location depends on your deployment method but if it's a raw windows install, probably the `artifacts` directory within the installation. All documented at https://docs.gocd.org/current/installation/install/server/windows.html

If your pipelines are slow, you probably want to look at the operating system process metrics (memory, CPU) where your agents are running, and compare to how long it takes if you run the same commands outside GoCD. Since GoCD agents just fork processes to run your builds, often slowness is not GoCD itself, but configuration of tooling or the server where the agents are running.

You might get wider help on the GoCD forums at https://groups.google.com/g/go-cd .

**Comment by ghost** [comment]
Thanks for the reply @chadlwilson 
Below are the details of build server 
Purpose: Build Sever Agent
OS: Windows Server 2019
Resource Requirements:
CPU(s): 16
Physical Memory : 64GB
Disk Size : 400 GB
GoCD Version: 22.2.0

when I run 3-4 pipelines it takes 2 hours to complete.
Hope above info is helpful.

**Reply by chadlwilson** [reply]
Please re-read the above? You will need to analyse OS metrics and npm build logs yourself, do some basic problem solving, and ask for more specific guidance based on your research. Generic hardware specs aren't helpful beyond noting that there there should be no issue with a properly configured GoCD server and agent working on that hardware. If agents are on the same host it depends what your builds are doing as to additional memory or CPU requirements.

No-one else can know what your npm build is doing, how long it normally should take outside GoCD or how long it takes on alternative hardware but yourself. 🙂
